### PR TITLE
Users/jack/ansible tests

### PR DESCRIPTION
**Pull request by jackjia-ibm**
Please check if your PR fulfills the following requirements. This is simply a reminder of what we are going to look for before merging your PR. If you don't know all of this information when you create this PR, don't worry. You can edit this template as you're working on it.

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Necessary documentation (if appropriate) have been added / updated
- [x] DCO signoffs have been added to all commits, including this PR

#### PR type
What type of changes does your PR introduce to Zowe? _Put an `x` in the box that applies to this PR. If you're unsure about any of them, don't hesitate to ask._ 
- [ ] Bugfix 
- [x] Feature 
- [ ] Other... Please describe:

#### Relevant issues

This is pre-req work required by:
- issue #517 
- issue #955 
- issue #712 
- issue https://github.com/zowe/zowe-install-test/issues/163

#### Changes proposed in this PR

- Using Ansible to drive Zowe installation,
- New test cases of installation, which allows extension in the future,
- Sanity tests are imported from zowe-install-test.
- zowe-install-test can be archived soon after this is merged.

#### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

#### Does this PR do something the person installing Zowe should know about?

---
* Affected function: _general area of interest_ *

---
* Description: _1 line description_ *

---
* Part: _name of customizable file involved_  *

---
_multi-line description_

#### Is there a related doc issue or Pull Request? 

Doc issue/PR number:

### CODE REVIEW

**Comment by jackjia-ibm** [comment]
Test run are successful with:
- testing convenient build installation: https://wash.zowe.org:8443/job/zowe-install-test-ansible/job/users%252Fjack%252Fansible-tests/29/
- testing SMPE FMID installation: https://wash.zowe.org:8443/job/zowe-install-test-ansible/job/users%252Fjack%252Fansible-tests/32/

**Comment by stevenhorsman** [comment]
@jackjia-ibm - did you mean to add us as reviewers rather than assign the PR to us? Is it ready now?

**Comment by jackjia-ibm** [comment]
@stevenhorsman yes it's ready for reviewing the first stage work, which is converting all functions we have on zowe-install-test.

The ongoing and future works will be:
- port the PTF install test to playbooks,
- add extended test scenarios (multiple instances of installation, multiple node.js version, acf2/ts)
- add plugin install and test.

File: playbooks/roles/zowe/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+# Variables should be verified and overwrittern.
+# ==============================================================================
+# temporary folder on remote
+work_dir_remote:
+# default zowe runtime root directory
+zowe_root_dir: ~/zowe
+# dataset prefix where zowe will be installed
+zowe_dataset_prefix:
+# userDir. This is replaced with instanceDir
+zowe_user_dir: ~/.zowe/user
```
Comment (stevenhorsman) [inline playbooks/roles/zowe/defaults/main.yml]: Same as above - do we still need this?

File: tests/installation/Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+    string(
+      name: 'ZOWE_CLI_ARTIFACTORY_BUILD',
+      description: 'Zowe artifactory download build',
+      defaultValue: 'Zowe CLI Bundle :: master',
+      trim: true
+    ),
+    // >>>>>>>> parametters for test cases
+    choice(
+      name: 'NODE_VERSION',
+      choices: ['v8.16.0', 'v6.17.0', 'v8.16.1', 'v8.16.2', 'v8.17.0', 'v12.13.0', 'v12.14.1'],
```
Comment (stevenhorsman) [inline tests/installation/Jenkinsfile]: 8.16.1 is officially not supported due to issues with it (https://docs.zowe.org/stable/troubleshoot/app-framework/app-known-issues.html#desktop-apps-fail-to-load), so maybe remove it from the list?

**Comment by jackjia-ibm** [comment]
Thanks a lot @stevenhorsman for the suggestions. I've resolved all of them and ran another 2 tests:

- convenience build: https://wash.zowe.org:8443/job/zowe-install-test-ansible/view/change-requests/job/PR-1193/2/
- fmid: https://wash.zowe.org:8443/job/zowe-install-test-ansible/view/change-requests/job/PR-1193/4/

which are both successful.

**Pull request merged**